user_config.c

```diff
@@ -0,0 +1,73 @@
+#include "user_config.h"
+#include <stdio.h>
+
+static uint8_t set_debug_level = 0, change_function = 0, temp_num = 0;
+static bool set_sensor_messages = false;
+static char save_type;
+
+// function takes in a message and parses it to see if max_debug_level() or allow_sensor_messages() needs to be changed
+void parse_usb_string(const char *input) {
+	uint8_t iterate = 0;
+	while (input[iterate] != '\0') {
+		switch (change_function) {
```

Comment:
So you're using change_function to represent the state of this parser. That's really good. The only problem is that the numbers have no semantic meaning, which makes it really hard to read. I think you want an enum type:

Something like this: https://stackoverflow.com/questions/15237656/using-enum-type-in-a-switch-statement

Reply:
I'm glad you commented on this because I knew that there was a better way but I couldn't come up with one at the time.